shared/src/main/java/com/couchbase/lite/internal/support/Log.java

```diff
@@ -385,49 +342,72 @@ public static void e(String tag, String formatString, Throwable tr, Object... ar
+        } catch (Exception e) {
+            msg = String.format(Locale.ENGLISH, "Unable to format log: %s (%s)", formatString, e.toString());
         }
+        sendToLoggers(LogLevel.ERROR, tag, msg);
     }
 
     public static void setLogLevel(LogDomain domain, LogLevel level) {
+        int actualLevel = level.equals(LogLevel.NONE) ? Log.NONE : Log.DEBUG;
         switch (domain) {
             case ALL:
-                enableLogging(DATABASE, level.getValue());
-                enableLogging(QUERY, level.getValue());
-                enableLogging(SYNC, level.getValue());
-                enableLogging(WEB_SOCKET, level.getValue());
-                enableLogging(C4LogDomain.BLIP, level.getValue());
-                enableLogging(C4LogDomain.SyncBusy, level.getValue());
+                enableLogging(DATABASE, actualLevel);
+                enableLogging(QUERY, actualLevel);
+                enableLogging(SYNC, actualLevel);
+                enableLogging(WEB_SOCKET, actualLevel);
+                enableLogging(C4LogDomain.BLIP, actualLevel);
+                enableLogging(C4LogDomain.SyncBusy, actualLevel);
                 break;
             case DATABASE:
-                enableLogging(DATABASE, level.getValue());
+                enableLogging(DATABASE, actualLevel);
                 break;
 
             case QUERY:
-                enableLogging(QUERY, level.getValue());
+                enableLogging(QUERY, actualLevel);
                 break;
 
             case REPLICATOR:
-                enableLogging(SYNC, level.getValue());
-                enableLogging(C4LogDomain.SyncBusy, level.getValue());
+                enableLogging(SYNC, actualLevel);
+                enableLogging(C4LogDomain.SyncBusy, actualLevel);
                 break;
 
             case NETWORK:
-                enableLogging(C4LogDomain.BLIP, level.getValue());
-                enableLogging(WEB_SOCKET, level.getValue());
+                enableLogging(C4LogDomain.BLIP, actualLevel);
+                enableLogging(WEB_SOCKET, actualLevel);
                 break;
         }
     }
 
     public static void enableLogging(String tag, int logLevel) {
-        // CBL logging
-        enabledTags.put(tag, logLevel);
         // LiteCore logging
         C4Log.setLevel(tag, logLevel);
     }
+
+    private static void sendToLoggers(LogLevel level, String tag, String msg) {
+        boolean fileSucceeded = false;
+        boolean consoleSucceeded = false;
+        try {
+            LogDomain domain = LogDomain.valueOf(tag);
+            Database.getLog().getFile().log(level, domain, msg);
+            fileSucceeded = true;
+            Database.getLog().getConsole().log(level, domain, msg);
+            consoleSucceeded = false;
+            Logger custom = Database.getLog().getCustom();
+            if(custom != null) {
+                custom.log(level, domain, msg);
+            }
+        } catch(Exception e) {
+            if(fileSucceeded) {
```

Comment:
I'm not sure if I understand this here. Should it be if (!fileSucceeded)?

Reply:
No, if fileSucceeded is false it means something is wrong with the file logger, and logging to it would cause another exception inside of the catch.  Basically, I want to log the error to all the loggers that are still working.